### PR TITLE
added a jax example

### PR DESCRIPTION
**Pull request by tanayvarshney**

### CODE REVIEW

File: Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    data_type: TYPE_FP32
+    dims: [-1, -1]
+  }
+]
+
+instance_group [
+  {
+    kind: KIND_GPU
+  }
+]
```
Comment (nealvaidya) [inline Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt:50]: Does this actually do anything when you use the python backend?

File: Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    name: "image"
+    data_type: TYPE_FP32
+    dims: [-1, -1, -1]
+  }
+]
+output [
+  {
+    name: "fc_out"
+    data_type: TYPE_FP32
+    dims: [-1, -1]
```
Comment (nealvaidya) [inline Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt:42]: This doesn't seem right. Shouldn't it be `dims: [1000]`?

File: Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+name: "resnet50"
+backend: "python"
+max_batch_size: 8
+
+input [
+  {
+    name: "image"
+    data_type: TYPE_FP32
+    dims: [-1, -1, -1]
```
Comment (nealvaidya) [inline Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt:35]: Does this model really work with images of any shape? Even in the channels/depth?

File: Quick_Deploy/JAX/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+Before diving into the specifics execution, an understanding of the underlying structure is needed. To use a JAX or a Flax model, the recommended path for this is using a ["Python Model"](https://github.com/triton-inference-server/python_backend#python-backend). Python models in Triton are basically classes with three Triton-specific functions: `initialize`, `execute` and `finalize`. Users can customize this class to serve any python function they write or any model they want as long as it can be loaded in python runtime. The `initialize` function runs when the python model is loaded into memory, and the `finalize` function runs when the model is unloaded from memory. Both of these functions are optional to define. For the purposes of this example, we will use the `initialize` and the `execute` functions to load and run(respectively) a `resnet18` model. 
+
+```
+class TritonPythonModel:
+    def initialize(self, args):
+        ...
```
Comment (nealvaidya) [inline Quick_Deploy/JAX/README.md]: Can you add some explanation of what you are putting in the config.pbtxt file? Mainly the input/output names and how they have to correspond to what's in the execute method

File: Quick_Deploy/JAX/model_repository/resnet50/1/model.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        self.resnet18 = fm.ResNet18(output='logits', pretrained='imagenet')
+
+
+    def execute(self, requests):
+        responses = []
+        for request in requests:
+            inp = pb_utils.get_input_tensor_by_name(request, "image")
+            input_image = inp.as_numpy()
+
+            params = self.resnet18.init(self.key, input_image)
```
Comment (nealvaidya) [inline Quick_Deploy/JAX/model_repository/resnet50/1/model.py:46]: I'm surprised by this `init` call here. I'm not overly familiar with how Flax or the Flaxmodels library structure things, but why do we call init here instead of in the `initialize` method. What does it actually do? Is it about needing to know the shape of the input image?

**Review by nealvaidya — CHANGES_REQUESTED** [review]

File: Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt (hunk 1 above)
Comment (tanayvarshney) [reply Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt:50]: Yes, it exposes a GPU I believe

File: Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt (hunk 2 above)
Comment (tanayvarshney) [reply Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt:42]: It [-1,1000] if I remember it correctly. I can re-run and check

File: Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt (hunk 3 above)
Comment (tanayvarshney) [reply Quick_Deploy/JAX/model_repository/resnet50/config.pbtxt:35]: https://github.com/matthias-wright/flaxmodels/blob/main/docs/Documentation.md#resnet18, seems so

File: Quick_Deploy/JAX/README.md (hunk 4 above)
Comment (tanayvarshney) [reply Quick_Deploy/JAX/README.md]: We have this info in other places, I am trying to avoid replication of information, but I see why it is needed. We should discuss more about this. We are bound to run into this issue later down the line

File: Quick_Deploy/JAX/model_repository/resnet50/1/model.py (hunk 5 above)
Comment (tanayvarshney) [reply Quick_Deploy/JAX/model_repository/resnet50/1/model.py:46]: Ah, good catch, I missed this.

**Comment by dyastremsky** [comment]
Pinging @tanayvarshney as a reminder, in case this is still in progress.
